dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAITextSearchService.cs

```diff
@@ -0,0 +1,217 @@
+        this.Attributes = new Dictionary<string, object?>
+        {
+            { "ServiceName", this._searchIndexClient.ServiceName },
+        };
+    }
+
+    /// <inheritdoc/>
+    async Task<KernelSearchResults<TextSearchResult>> ITextSearchService<TextSearchResult>.SearchAsync(string query, SearchExecutionSettings? searchSettings, Kernel? kernel, CancellationToken cancellationToken)
+    {
+        return await this.ExecuteSearchAsync<TextSearchResult>(query, searchSettings, kernel, cancellationToken).ConfigureAwait(true);
+    }
+
+    /// <inheritdoc/>
+    async Task<KernelSearchResults<SearchDocument>> ITextSearchService<SearchDocument>.SearchAsync(string query, SearchExecutionSettings? searchSettings, Kernel? kernel, CancellationToken cancellationToken)
+    {
+        return await this.ExecuteSearchAsync<SearchDocument>(query, searchSettings, kernel, cancellationToken).ConfigureAwait(true);
+    }
+
+    #region private
+
+    private readonly SearchIndexClient _searchIndexClient;
+    private readonly string? _index;
+
+    /// <summary>
+    /// Perform a search for content related to the specified query.
+    /// </summary>
+    /// <param name="query">What to search for</param>
+    /// <param name="searchSettings">Option search execution settings</param>
+    /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    private async Task<KernelSearchResults<T>> ExecuteSearchAsync<T>(string query, SearchExecutionSettings? searchSettings, Kernel? kernel, CancellationToken cancellationToken) where T : class
+    {
+        Verify.NotNullOrWhiteSpace(query);
+        Verify.NotNull(searchSettings);
+
+        var index = searchSettings.Index ?? this._index;
+        Verify.NotNullOrWhiteSpace(index);
+
+        var indexName = this.NormalizeIndexName(index);
+        var searchClient = this._searchIndexClient.GetSearchClient(indexName);
+
+        var azureSearchSettings = AzureAISearchExecutionSettings.FromExecutionSettings(searchSettings);
+
+        //SearchResults<T>? searchResults = null;
+        try
+        {
+            var response = await searchClient.SearchAsync<SearchDocument>(query, azureSearchSettings.SearchOptions, cancellationToken).ConfigureAwait(true);
+            SearchResults<SearchDocument>? searchResults = response.Value;
+            if (typeof(T) == typeof(TextSearchResult))
+            {
+                return new KernelSearchResults<T>(searchResults, (IAsyncEnumerable<T>)this.GetResultsAsync(searchResults, azureSearchSettings.NameField, azureSearchSettings.ValueField, azureSearchSettings.LinkField, cancellationToken), searchResults?.TotalCount, GetResultsMetadata(searchResults));
+            }
+            if (typeof(T) == typeof(SearchDocument))
```

Comment:
I think there is no need to define one `ExecuteSearchAsync` method, which will be re-usable, but will do type checking inside. Instead, we can just place this logic to `ITextSearchService<TextSearchResult>.SearchAsync` and `ITextSearchService<SearchDocument>.SearchAsync` methods without type checking.